### PR TITLE
Feature favorites recipes

### PR DESCRIPTION
**Pull request by thameursaadi**
@SamiDghim @melas123 @medBelaid

### CODE REVIEW

File: app/models/recipe.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@ class Recipe < ActiveRecord::Base
   has_many :ingredient_recipes
   has_many :images
   has_many :favorites, :dependent => :destroy
```
Comment (SamiDghim) [inline app/models/recipe.rb:9]: plz update your ruby syntax

Comment (thameursaadi) [inline app/models/recipe.rb:9]: done

File: app/controllers/profile_controller.rb [hunk 2]
```diff
@@ -9,4 +9,12 @@ def index
       respond_with( Paginator.pagination_attributes( @recipesUser ).merge!( recipes:  @recipesUser ), status: 200 )
     end
   end
+  #  get "profile/favorite_list"
+  def favorite_list
+    @favorites=Recipe.where( id: current_user.favorites.pluck( :recipe_id ) )
+  end
+  #  post "profile/destroy_favorite"
+  def destroy_favorite
+    (Favorite.where( id: params[:favorite_id], user_id: current_user.id )).first.destroy
```
Comment (achr3f) [inline app/controllers/profile_controller.rb]: no need to the first `()`

> Favorite.where( id: ..... ).first.destroy

File: app/controllers/profile_controller.rb [hunk 3]
```diff
@@ -9,4 +9,12 @@ def index
       respond_with( Paginator.pagination_attributes( @recipesUser ).merge!( recipes:  @recipesUser ), status: 200 )
     end
   end
+  #  get "profile/favorite_list"
+  def favorite_list
+    @favorites=Recipe.where( id: current_user.favorites.pluck( :recipe_id ) )
```
Comment (achr3f) [inline app/controllers/profile_controller.rb]: space between `=`

File: app/models/recipe.rb [hunk 4]
```diff
@@ -5,8 +5,9 @@ class Recipe < ActiveRecord::Base
   belongs_to :user
   has_many :ingredient_recipes
   has_many :images
-  has_many :favorites, :dependent => :destroy
-  has_many :comments, :dependent => :destroy
+  has_many :favorites, dependent: :destroy
+
```
Comment (achr3f) [inline app/models/recipe.rb]: remove the empty line

File: app/models/user.rb [hunk 5]
```diff
@@ -3,9 +3,12 @@ class User < ActiveRecord::Base
...
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
-  has_many :recipes, :dependent => :destroy
-  has_many :favorites, :dependent => :destroy
-  has_many :comments, :dependent => :destroy
+
+
+  has_many :recipes, dependent: :destroy
+  has_many :favorites, dependent: :destroy
+
```
Comment (achr3f) [inline app/models/user.rb]: same  remove the empty line

File: db/schema.rb [hunk 6]
```diff
@@ -13,6 +13,8 @@
 
 ActiveRecord::Schema.define(version: 20160126095608) do
 
+
```
Comment (achr3f) [inline db/schema.rb]: why add two empty line ???

File: app/controllers/profile_controller.rb [hunk 7]
```diff
@@ -9,4 +9,12 @@ def index
       respond_with( Paginator.pagination_attributes( @recipesUser ).merge!( recipes:  @recipesUser ), status: 200 )
     end
   end
+  #  get "profile/favorite_list"
+  def favorite_list
+    @favorites = Recipe.where( id: current_user.favorites.pluck( :recipe_id ) )
```
Comment (SamiDghim) [inline app/controllers/profile_controller.rb]: plz use rails scope

Comment (thameursaadi) [inline app/controllers/profile_controller.rb]: done

**Pull request closed without merging**